sympy/printing/jscode.py

```diff
@@ -111,6 +111,18 @@ def _print_Rational(self, expr):
         p, q = int(expr.p), int(expr.q)
         return '%d/%d' % (p, q)
 
+    def _print_Mod(self, expr):
+        num, den = expr.args
+        PREC = precedence(expr)
+        snum, sden = [self.parenthesize(arg, PREC) for arg in expr.args]
+        # % is remainder (same sign as numerator), not modulo (same sign as
+        # denominator), in js. Hence, % only works as modulo if both numbers
+        # have the same sign
+        if (num.is_nonnegative and den.is_nonnegative or
+            num.is_nonpositive and den.is_nonpositive):
+         return f"({snum} % {sden})"
+        return f"((({snum} % {sden}) + {sden}) % {sden})"
```

Comment:
do we need the explicit extra parenthesis? I'm leaning towards that we should just trust that `_print_WhateverOperator` actually calls `parenthesize` just like we are doing here.

Reply:
Maybe not. Strictly, the `({snum} % {sden})` should never be needed in any of the expressions. For the latter it was more that the expression I saw was printed like this, so it looked "better" somehow. The outer in the second one should not be needed either. It was more because of the issues with mod elsewhere that I thought it was safer to have them there since there is limited harm in generated code, especially compiled.

Now that #22080 is merged, I can revisit it and see if double parenthesis show up anywhere.

Reply:
Seems like #22080 changed the behaviour a bit (to the better). So with added tests, I am quite confident in removing the outer parenthesis.